api/src/apps/routes.rs

```diff
@@ -54,42 +70,100 @@ fn apps(
     request_info: RequestInfo,
     host_meta_cache: State<HostMetaCache>,
 ) -> Result<Json<MultiMap<String, Service>>, HttpApiProblem> {
-    let services = apps.get_apps()?;
+    let services = runtime.block_on(apps.get_apps())?;
     Ok(Json(
         host_meta_cache.update_meta_data(services, &request_info),
     ))
 }
 
+#[get("/<app_name>/status-changes/<status_id>", format = "application/json")]
+fn status_change(
+    app_name: Result<AppName, AppNameError>,
+    status_id: Result<AppStatusChangeId, AppStatusChangeIdError>,
+    apps: State<Arc<Apps>>,
+    options: RunOptions,
+) -> Result<AsyncCompletion<Json<Vec<Service>>>, HttpApiProblem> {
+    let app_name = app_name?;
+    let status_id = status_id?;
+
+    let apps = apps.clone();
+    let future = async move { apps.wait_for_status_change(&status_id).await };
```

Comment:
These `async move { … }` are not necessary and you can just write:

```rust
let future = apps.wait_for_status_change(&status_id);
```

In this file are two more blocks that do not need an `async move`.

Reply:
But then the `&status_id`, owned by the route-fn, might be dropped before `wait_for_status_change` completes. I mean, I could `.clone()` the `status_id`, but is that an improvement? (I'm asking because I don't know :D)

Reply:
Rust's pinning ensures that for you. The [async-book states the same question](https://rust-lang.github.io/async-book/04_pinning/01_chapter.html):

> However, what happens if we have an async block that uses references?

This reference explains it quite well what is done under `rustc`'s hood